diff --git a/lib/index.js b/lib/index.js
index 2140eb6fd337be8166332906fb047e878ab23604..9e32d6af45bdf1502472c444886423cc8ce3b7a1 100644
--- a/lib/index.js
+++ b/lib/index.js
@@ -37,10 +37,18 @@ function adminOnlyWrapper(error, orig, args) {
     name: null,
     roles: ["_admin"]
   };
-  if (userCtx.roles.indexOf("_admin") !== -1) {
+  // NOTE: Edited because db seems to be undefined if we are in admin party mode.
+  if (userCtx.roles.indexOf("_admin") !== -1 || !(args.options || {}).userCtx) {
     //server admin can do everything
     return orig();
   }
+  // Seems to be the case when no security object has been applied yet.
+  if (!args.db)
+    throw new PouchDBPluginError({
+      status: 404,
+      name: "not_found",
+      message: "No database available."
+    });
   return args.db.getSecurity().then(function (security) {
     var dbAdmins = (security.admins || {});
     var isDbAdmin = (
